CONTRIBUTING.md

```diff
@@ -0,0 +1,63 @@
+# Contributing To Trow
+
+Firstly, big thanks for considering contributing to the project. We really hope to make this into a
+community project and to do that we need your help!
+
+## Code of Conduct
+
+We are serious about making this a welcoming, happy project. We will not tolerate discrimination,
+aggressive or insulting behaviour.
+
+To this end, the project and everyone participating in it is bound by the [Code of
+Conduct](CODE_OF_CONDUCT.md). By participating, you are expected to uphold this code. Please report
+unacceptable behaviour to any of the project admins or adrian.mouat@container-solutions.com.
+
+## Bugs
+
+If you find a bug, please [open an issue](https://github.com/ContainerSolutions/trow/issues)! Do try
+to include all the details needed to recreate your problem. This is likely to include:
+
+ - The version of Trow being used
+ - How Trow was installed
+ - The exact platform and version of the platform that Trow is running on
+ - The steps taken to cause the bug
+
+## Building Features and Documentation
+
+If you're looking for something to work on, take look at the issue tracker, in particular any items
+labelled [good first issue](https://github.com/ContainerSolutions/trow/labels/good%20first%20issue).
+Please leave a comment on the issue to mention that you have started work, in order to avoid
+multiple people working on the same issue.
+
+If you have an idea for a feature - whether or not you have time to work on it - please also open an
+issue describing your feature and label it "enhancement". We can then discuss it as a community and
+see what can be done. Please be aware that some features may not align with the project goals and
+might therefore be closed. In particular, please don't start work on a new feature without
+discussing it first to avoid wasting effort. We do commit to listening to all proposals and will do
+our best to work something out!
+
+Once you've got the go ahead to work on a feature, you can start work. Feel free to communicate with
+team via updates on the issue tracker and ask for feedback, pointers etc. Once you're happy with
+your code, go ahead and open a Pull Request.
+
+## Pull Request Process
+
+Most of the code is written in Rust and uses the standard style guidelines - please run your code
```

Comment:
My suggestion would be to add a Makefile, and or a Dockerfile to test the project in a repeatable way. Since you're using latest, this could vary by person depending on when they last updated. Pinning this to a version, in a Dockerfile, and having a not so complex Makefile to drive executing that locally, could be useful. I could contribute this if you're interested.

Reply:
I loathe Make with a passion :)

There are Dockerfiles, including a Dockerfile that runs the tests. I'll try to be a bit clearer in the text.

Reply:
Oh Adrian, you were off to such a great start, and then... :grimacing: :laughing: 

Come to the dark side, Make can be your friend :wink:

Reply:
I should mention that it's difficult to pin to a version of nightly - eventually you end with a dependency that doesn't work. So what we try to say is "the latest nightly that supports rustfmt and bunch of other components". It's a mess. The solution is moving to stable rather than pinning.

Reply:
I wouldn't be against Make if it did something useful for us, but at the moment it would just be a complicated way to call single line commands.